File: objective-c/iOS13Display/iOS13Display/ViewControllers/VCMain.m
//
//  VCMain.m
//  iOS13Display
//
//  Created by Alexandr Kirilov on 08/12/2019.
//  Copyright © 2019 Alexandr Kirilov. All rights reserved.
//

#import "VCMain.h"

@interface VCMain ()

@property (strong,nonatomic) UIButton *pButtonExit;
@property (strong,nonatomic) UILabel *pLabelLanguage;
@property (strong,nonatomic) UILabel *pLabelNotice;
@property CGFloat pScreenWidth;
@property CGFloat pScreenHeight;

@end

@implementation VCMain

-(void) loadView {
	
	[super loadView];
	self.view.backgroundColor = _COLOR_WHITE;
}

-(void) viewDidLoad {
	
    [super viewDidLoad];
	_pScreenWidth = CGRectGetWidth(self.view.bounds);
	_pScreenHeight = CGRectGetHeight(self.view.bounds);
	
	[self mCreateButtonExit];
	[self mCreateLabelLanguage];
	[self mCreateLabelNotice];
}


#pragma mark UI Elements

-(void) mCreateButtonExit {

	_pButtonExit = [[UIButton alloc] init];
	_pButtonExit.backgroundColor = _COLOR_BLUE_DARK;
	[_pButtonExit setTitleColor:_COLOR_WHITE_CLEAN forState:UIControlStateNormal];
	[_pButtonExit setTitle:@"Exit" forState:UIControlStateNormal];
	[_pButtonExit addTarget:self action:@selector(mDoExit) forControlEvents:UIControlEventTouchUpInside];
	[self.view addSubview:_pButtonExit];
	_pButtonExit.translatesAutoresizingMaskIntoConstraints = NO;
	[_pButtonExit.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor constant:-_pScreenHeight * 0.05].active = YES;
	[_pButtonExit.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor].active = YES;
	[_pButtonExit.widthAnchor constraintEqualToConstant:_pScreenWidth * 0.8].active = YES;
	[_pButtonExit.heightAnchor constraintEqualToConstant:_pScreenHeight * 0.075].active = YES;
}

-(void) mCreateLabelLanguage {

	_pLabelLanguage = [[UILabel alloc] init];
	_pLabelLanguage.text = @"Objective-C Examples";
	_pLabelLanguage.textAlignment = NSTextAlignmentCenter;
	[self.view addSubview:_pLabelLanguage];
	_pLabelLanguage.translatesAutoresizingMaskIntoConstraints = NO;
	[_pLabelLanguage.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor].active = YES;
	[_pLabelLanguage.topAnchor constraintEqualToAnchor:self.view.topAnchor constant:_pScreenHeight * 0.05].active = YES;
	[_pLabelLanguage.heightAnchor constraintEqualToConstant:_pScreenHeight * 0.05].active = YES;
}

-(void) mCreateLabelNotice {
	
	_pLabelNotice = [[UILabel alloc] init];
	_pLabelNotice.text = @"Disable DarkMode and Invert colors";
	_pLabelNotice.textAlignment = NSTextAlignmentCenter;
	[self.view addSubview:_pLabelNotice];
	_pLabelNotice.translatesAutoresizingMaskIntoConstraints = NO;
	[_pLabelNotice.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor].active = YES;
	[_pLabelNotice.topAnchor constraintEqualToAnchor:_pLabelLanguage.bottomAnchor].active = YES;
	[_pLabelNotice.heightAnchor constraintEqualToConstant:_pScreenHeight * 0.05].active = YES;
}


#pragma mark UI Elements Actions

-(void) mDoExit {

	[HMain mDoExit];
}

@end

File: swift/UIKit/UIKit UINavigationController/UIKit UINavigationController/ViewControllers/VCMain.swift
//
//  VCMain.swift
//  UIKit UINavigationController
//
//  Created by Alexandr Kirilov on 09/08/2019.
//  Copyright © 2019 Arboreus. All rights reserved.
//

import UIKit

class VCMain: UIViewController {

	let pLabelLanguage: UILabel = UILabel();
	let pLabelNotice: UILabel = UILabel();
	let pButtonExit: UIButton = UIButton();

	override func loadView() {
		
		super.loadView();
		self.view.backgroundColor = _COLOR_WHITE;
	}
	
	override func viewDidLoad() {
		
		super.viewDidLoad();
		
		self.mCreateLabelLanguage();
		self.mCreateLabelNotice();
		self.mCreateButtonExit();
	}
	
	private func mCreateLabelLanguage() -> Void {

		pLabelLanguage.text = "Swift examples";
		self.view.addSubview(pLabelLanguage);
		pLabelLanguage.translatesAutoresizingMaskIntoConstraints = false;
		pLabelLanguage.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 50).isActive = true;
		pLabelLanguage.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true;
	}

	private func mCreateLabelNotice() -> Void {

		pLabelNotice.text = "UINavigationController Example";
		self.view.addSubview(pLabelNotice);
		pLabelNotice.translatesAutoresizingMaskIntoConstraints = false;
		pLabelNotice.topAnchor.constraint(equalTo: self.pLabelLanguage.bottomAnchor, constant: 20).isActive = true;
		pLabelNotice.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true;
	}
	
	private func mCreateButtonExit() -> Void {

		self.view.addSubview(pButtonExit);
		pButtonExit.translatesAutoresizingMaskIntoConstraints = false;
		pButtonExit.backgroundColor = _COLOR_BLUE_DARK;
		pButtonExit.setTitleColor(.white, for: .normal);
		pButtonExit.setTitle("Exit", for: .normal);
		pButtonExit.addTarget(self, action: #selector(mDoExit), for: .touchUpInside);
		pButtonExit.centerXAnchor.constraint(
			equalTo: self.view.centerXAnchor
		).isActive = true;
		pButtonExit.bottomAnchor.constraint(
			equalTo: self.view.bottomAnchor,
			constant: -self.view.bounds.height * 0.06
		).isActive = true;
		pButtonExit.widthAnchor.constraint(
			equalTo: self.view.widthAnchor,
			constant: -self.view.bounds.width * 0.2
		).isActive = true;
		pButtonExit.heightAnchor.constraint(
			equalTo: self.view.heightAnchor,
			constant: -self.view.bounds.height * 0.93
		).isActive = true;
	}
	
	@objc private func mDoExit() -> Void {

	   	HMain.mDoExit();
	}
}
